hydromt/config.py

```diff
@@ -3,15 +3,16 @@
 """
 config functions
 """
-
-import codecs
-from configparser import ConfigParser
-from os.path import dirname, join, abspath, exists
-from ast import literal_eval
-import numpy as np
-import abc
+from os.path import dirname, join, abspath, exists, splitext
 from pathlib import Path
 from typing import Union, Dict, List
+from ast import literal_eval
+import codecs
+import abc
+from configparser import ConfigParser
+from warnings import warn
+
+import yaml
 
 
 def configread(
```

Comment:
This function has become a bit hard to read. I've splitted the elements into multiple functions to make it more clear what happens.

Reply:
Thanks! Looks good to me.